obspy/core/util/decorator.py

```diff
@@ -279,10 +280,17 @@ def new_func(*args, **kwargs):
             # check kwargs
             if arg_kwarg_name in kwargs:
                 if isinstance(kwargs[arg_kwarg_name], (str, native_str)):
-                    if kwargs[arg_kwarg_name].startswith(prefix):
+                    arg_ = kwargs[arg_kwarg_name]
+                    if isinstance(arg_, (bytes, native_bytes)):
+                        # try to find encoding specified in document
+                        match = re.match(r'[^>]* encoding=(["\'])(.+?)\1',
+                                         arg_)
```

Comment:
Well, then the behavior is unchanged.. this was prompted by a xml file getting returned encoded.

I don't know if we want to start trying to extract encoding information in all kinds of ways, unless we actually encounter a problem.. I could decode with default of UTF-8 in those cases?